tests/RouteTest.php

```
@@ -255,6 +279,52 @@ public function testMiddlewareAfterAction(): void
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Invalid $middlewareDefinitions provided, list of string or array or callable expected.');
+
+        new Route([Method::GET], '/', middlewares: ['ValidString', (object) ['test' => 1]]);
+    }
+
+    public function testInvalidMiddlewares(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Invalid $middlewareDefinitions provided, list of string or array or callable expected.');
+
+        $route = new Route([Method::GET], '/', middlewares: [static fn() => new Response(), (object) ['test' => 1]]);
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$route'. (undefined)

(UnusedLocalVariable)